pkg/driver/node_test.go

```diff
@@ -2414,16 +2445,52 @@ func TestRemoveNotReadyTaint(t *testing.T) {
 			setup: func(t *testing.T, mockCtl *gomock.Controller) func() (kubernetes.Interface, error) {
 				t.Setenv("CSI_NODE_NAME", nodeName)
 				getNodeMock, mockNode := getNodeMock(mockCtl, nodeName, &corev1.Node{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: nodeName,
+					},
 					Spec: corev1.NodeSpec{
 						Taints: []corev1.Taint{
 							{
 								Key:    AgentNotReadyNodeTaintKey,
-								Effect: "NoExecute",
+								Effect: corev1.TaintEffectNoExecute,
 							},
 						},
 					},
 				}, nil)
-				mockNode.EXPECT().Patch(gomock.Any(), gomock.Eq(nodeName), gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, fmt.Errorf("Failed to patch node!"))
+
+				storageV1Mock := NewMockStorageV1Interface(mockCtl)
+				getNodeMock.(*MockKubernetesClient).EXPECT().StorageV1().Return(storageV1Mock).AnyTimes()
+
+				csiNodesMock := NewMockCSINodeInterface(mockCtl)
+				storageV1Mock.EXPECT().CSINodes().Return(csiNodesMock).Times(1)
+
+				count := int32(1)
+				mockCSINode := &v1.CSINode{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: nodeName,
+					},
+					Spec: v1.CSINodeSpec{
+						Drivers: []v1.CSINodeDriver{
+							{
+								Name:   DriverName,
+								NodeID: nodeName,
+								Allocatable: &v1.VolumeNodeResources{
+									Count: &count,
+								},
+							},
+						},
+					},
+				}
+
+				csiNodesMock.EXPECT().
```

Comment:
I know that I am the one responsible for it, and I'm not asking you to fix what's already broken in this PR.

But in the future, we really need to move from this boilerplate disaster to using the fake client implementation.